lib/src/cmaf/ham/types/mapper/dash/AdaptationSet.ts

```diff
@@ -0,0 +1,42 @@
+import type { AudioChannelConfiguration } from './AudioChannelConfiguration.js';
+import type { ContentComponent } from './ContentComponent.js';
+import type { Representation } from './Representation.js';
+import type { Role } from './Role.js';
+import type { SegmentList } from './SegmentList.js';
+import type { SegmentTemplate } from './SegmentTemplate.js';
+
+/**
+ * DASH Adaptation Set
+ *
+ * @group CMAF
+ * @alpha
+ */
+export type AdaptationSet = {
+	$: {
```

Comment:
Why do all of these types have a `$` property? I assume it's the XML node's attributes list, but I'd rather not have users make assumptions about the code.

Reply:
Yes, your assumption is correct according to the xml2js documentation. The $ object in the output from xml2js (https://www.npmjs.com/package/xml2js) represents the attributes of the XML node. For example, in XML like:

```
<person name="John" age="30">
  <phone>123456</phone>
</person>
```

xml2js would convert it into a JavaScript object where the $ object contains the attributes of the <person> 

```
{
  person: {
    $: { name: "John", age: "30" },
    phone: "123456"
  }
}
```

In this implementation, DashManifest (and all its sub-parts) is tightly coupled with xml2js. Perhaps we could leave a comment or something to that effect. What do you think @littlespex ?

Reply:
The tight coupling is something we will need to address at some point. I know dash.js specifically moved away from xml2json, and we want CML to be as dependency free as possible. We can leave this in place for now while we come up with a better solution.

Reply:
Agree, I think we should think about using stuff from dash.js to parse the Dash Manifest and maybe that could be a contribution for dash.js also